src/main/java/app/servlets/ControllerServlet.java

```diff
@@ -0,0 +1,232 @@
+                                }
+                                if (name.equals("pages")) {
+                                    book.setPages(Integer.parseInt(value));
+                                }
+                                if (name.equals("isbn")) {
+                                    book.setIsbn(Long.parseLong(value));
+                                }
+                                if (name.equals("total_amount")) {
+                                    book.setAmount(Integer.parseInt(value));
+                                }
+                            } else {
+                                filename = new File(item.getName()).getName();
+                                int i = filename.lastIndexOf('.');
+                                if (i != -1) {
+                                    String filename2 = filename.substring(0, i) + id1;
+                                    id1++;
+                                    String ext = filename.substring(i);
+                                    filename = filename2 + ext;
+                                }
+                                try {
+                                    File file = new File(UPLOAD_DIRECTORY + File.separator + filename);
+                                    item.write(file);
+                                    byte[] bArray = readFileToByteArray(file);
+                                    String base64Image = Base64.encodeBase64String(bArray);
+                                    book.setCover64(base64Image);
+                                } catch (Exception e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                        }
+                    }
+                    id++;
+                    book.setId(id);
+                    book.setStatus(st_1);
+
+                    insertBook(req, resp, book);
+                    listBook(req, resp);
+                    break;
+                case "/remove":
+                    deleteBook(req, resp);
+                    break;
+                case "/edit":
+                    showEditForm(req, resp);
+                    break;
+                case "/update":
+                    updateBook(req, resp);
+                    break;
+                case "/list":
+                default:
+                    listBook(req, resp);
+                    break;
+            }
+
+        } catch (SQLException ex) {
+            throw new ServletException(ex);
+        } catch (DBException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void insertBook(HttpServletRequest req, HttpServletResponse resp, Book book) throws SQLException, IOException, ServletException {
+        try {
+            service.addBook(book);
+            req.setAttribute("book", book);
+        } catch (DBException e) {
+            System.out.println("Cannot add book in the database");
```

Comment:
Log error with exception: in case something bad happens to your server in production, you wont be able to restart your server fast in debug mode (you could attach a remote debug, but I doubt that anyone would leave a debug port open for productions, since it is very dangerous). Logs are your friends. Especcialy if someone reports a bug, that happened 'somewhere between 3 and 4 am'. In your case you would resort to hundreds of logs and 'souts', but with a logger.error, you could 'grep' all logs between interval of time or even dates